core/src/main/java/org/opensearch/sql/expression/HighlightExpression.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.expression;
+
+import java.util.List;
+import lombok.Getter;
+import org.opensearch.sql.common.utils.StringUtils;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.data.type.ExprCoreType;
+import org.opensearch.sql.data.type.ExprType;
+import org.opensearch.sql.expression.env.Environment;
+import org.opensearch.sql.expression.function.BuiltinFunctionName;
+
+/**
+ * Highlight Expression.
+ */
+@Getter
+public class HighlightExpression extends FunctionExpression {
+  private final Expression highlightField;
+
+  /**
+   * HighlightExpression Constructor.
+   * @param highlightField : Highlight field for expression.
+   */
+  public HighlightExpression(Expression highlightField) {
+    super(BuiltinFunctionName.HIGHLIGHT.getName(), List.of(highlightField));
+    this.highlightField = highlightField;
+  }
+
+  /**
+   * Return String or Map value matching highlight field.
+   * @param valueEnv : Dataset to parse value from.
+   * @return : String or Map value of highlight fields.
+   */
+  @Override
+  public ExprValue valueOf(Environment<Expression, ExprValue> valueEnv) {
+    String refName = "_highlight" + "." + StringUtils.unquoteText(getHighlightField().toString());
+    return valueEnv.resolve(DSL.ref(refName, ExprCoreType.STRING));
+  }
+
+  /**
+   * Get type for HighlightExpression.
+   * @return : String type.
+   */
+  @Override
+  public ExprType type() {
```

Comment:
@forestmvey, can you change the return value here to match `ExprValue.type()` returned by `HighlightExpression::valueOf`.

`JDBCFormatter` uses `HighlightExpression::type` to construct the schema object in the response JSON. At the moment it's always `STRING` for highlight fragments.

With `highlight(*)` in mind it may be better to always return `ExprTupleValue`. That way if the user changes their statement from highlighting one field to highlighting many, they will not need to change the code that examines the response.